libs/meta/include/meta/log2.hpp

```diff
@@ -17,18 +17,22 @@
 //
 //------------------------------------------------------------------------------
 
-#include "vectorise/meta/log2.hpp"
+#include "meta/type_traits.hpp"
 
 namespace fetch {
 namespace meta {
 
-template <uint64_t VALUE>
-struct IsLog2
+template <typename T>
+constexpr EnableIf<IsInteger<T>, bool> IsLog2(T val) noexcept
```

Comment:
Does adding noexcept to a constexpr do anything?

Reply:
Yes, because `constexpr` **function** can be called in both contexts - compile-time and run-time. 
The `noexcept` is for when this function is called in run-time.

Reply:
This was my original suggestion (please check syntax)

```suggestion
constexpr IfIsInteger<T, bool> IsLog2(T val) noexcept
```